File: BECore/app/model/Company.js
/// <summary>
/// Model EXTJS Company.
/// </summary>
Ext.define('QLDT.model.Company', {
    extend: 'Ext.data.Model',

    idProperty: 'CompanyID',
    idType: 'Guid',

    code: 'CompanyCode',
    name: 'CompanyName',

    fields: [
		{ name: "CompanyID", type: "string" },
		{ name: "CompanyCode", type: "string" },
		{ name: "CompanyName", type: "string" },
		{ name: "CompanyShortName", type: "string" },
		{ name: "Tel", type: "string" },
		{ name: "Fax", type: "string" },
		{ name: "Email", type: "string" },
		{ name: "Address", type: "string" },
		{ name: "Website", type: "string" },
		{ name: "Director", type: "string" },
		{ name: "AccountNumber", type: "string" },
		{ name: "BankName", type: "string" },
		{ name: "Description", type: "string" },
		{ name: "IsLeaf", type: "boolean" },
		{ name: "Inactive", type: "boolean" },
		{ name: "SortOrder", type: "int" },
		{ name: "CreatedDate", type: "date" },
		{ name: "ModifiedDate", type: "date" },
		{ name: "IPAddress", type: "string" },
		{ name: "ModifiedBy", type: "string" },
		{ name: "CreatedBy", type: "string" },
        { name: "Owned", type: "boolean" },
    ]
});

File: MT.Model/Dictionary/Company.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace MT.Model
{
    #region Company:EntityBase
    /// <summary>
    /// This object represents the properties and methods of a Company:EntityBase.
    /// </summary>
    public class Company : EntityBase
    {
        [Key]
        public Guid CompanyID { get; set; }
        public Guid ParentID { get; set; }
        public string CompanyCode { get; set; }
        public string CompanyName { get; set; }
        public string CompanyShortName { get; set; }
        public string Tel { get; set; }
        public string Fax { get; set; }
        public string Email { get; set; }
        public string Address { get; set; }
        public string Website { get; set; }
        public string Director { get; set; }
        public string AccountNumber { get; set; }
        public string BankName { get; set; }
        public string Description { get; set; }
        public bool IsLeaf { get; set; }
        public bool Inactive { get; set; }
        public int SortOrder { get; set; }
        public bool Owned { get; set; }
    }
    #endregion
}

